### ISSUE TITLE
VoC option honk_and_blink

### ISSUE THREAD

**Issue by FireWizard52** [molobrakos/volvooncall#42]
I tested all options, after mqtt issue has been solved and discovered the following when:

pi@rpi3plus:~/.local/bin $ voc honk_and_blink
/home/pi/.local/lib/python3.7/site-packages/volvooncall/volvooncall.py:355: RuntimeWarning: coroutine 'Vehicle.call' was never awaited
  self.call("honkAndBlink")
RuntimeWarning: Enable tracemalloc to get the object allocation traceback
Traceback (most recent call last):
  File "/home/pi/.local/bin/voc", line 296, in <module>
    run(main(args), debug=debug)
  File "/usr/local/lib/python3.7/asyncio/runners.py", line 43, in run
    return loop.run_until_complete(main)
  File "/usr/local/lib/python3.7/asyncio/base_events.py", line 584, in run_until_complete
    return future.result()
  File "/home/pi/.local/bin/voc", line 246, in main
    await vehicle.honk_and_blink()
TypeError: object NoneType can't be used in 'await' expression

Any idea?

**Comment by molobrakos** [comment]
That is a bug, thanks! will fix tomorrow (unless someone else submits a pr)

**Comment by molobrakos** [comment]
Fix: https://github.com/molobrakos/volvooncall/commit/be9dab8984de10bbc2091ad9e975bfa5ffefd555

**Comment by FireWizard52** [comment]
Hi,

I tested your bugfix and can confirm that no more error messages are given.
However, the desired action is not carried out.
See below:
pi@rpi3plus:~ $ voc -v honk_and_blink
19-01-21 16:02.10  INFO (MainThread) [volvooncall.volvooncall] volvooncall.volvooncall 0.8.8 /home/pi/.local/lib/python3.7/site-packages/volvooncall/volvooncall.py
19-01-21 16:02.10  INFO (MainThread) [volvooncall.volvooncall] Updating
19-01-21 16:02.10  INFO (MainThread) [volvooncall.volvooncall] Querying vehicles
19-01-21 16:02.10 WARNING (MainThread) [volvooncall.volvooncall] Failure when communcating with the server: 404, message='Not Found'
19-01-21 16:02.10 WARNING (MainThread) [volvooncall.volvooncall] Failure to execute: 404, message='Not Found'

Using the Android app works fine.

Would it be possible also to include this option im MQTT

**Comment by molobrakos** [comment]
do you get `"honkAndBlinkSupported": true` from `./voc print`?

**Comment by FireWizard52** [comment]
Hi Erik,

Yes it is supported.

pi@rpi3plus:~ $ voc print|grep honk
    "honkAndBlinkDistance": 500,
    "honkAndBlinkSupported": true,
    "honkAndBlinkVersionsSupported": [
        "honkAndOrBlink",
        "honkAndBlink"

**Comment by molobrakos** [comment]
Please see [this comment](https://github.com/molobrakos/volvooncall/issues/7#issuecomment-392582296)

**Comment by FireWizard52** [comment]
Indeed it makes sense that you have to submit your location in order to be able to hear or see it
Thanks for your reply.

**Comment by FireWizard52** [comment]
Hello Erik,

After I got a new car some weeks ago, I decided to look once again to this issue.
Although some developers claim that the function honk and blink is supported I only see posts that users can't get it to work. However most users do not pay much attention to this function, but it would be nice to make it work. Some other developments are all based on your script, as you might know. So I don't think their claims are correct.

I tested it with, what I believe, all possible combinations of honk_and_blink and/or call but it didn't function.

So I decided to user a capture app on the phone and to capture the data send from the VoC app to the server.

Regarding honk and blink see the results below:
-------------------------------------------------------------------------------------------------------
X-Client-Version: 4.4.5.21126
X-Device-Id: 55792eff9c2ac026
X-OS-Type: Android
X-App-Name: Volvo On Call
X-OS-Version: 23
X-Originator-Type: app
Accept: application/vnd.wirelesscar.com.voc.Service.v4+json; charset=utf-8
X-Request-Id: 436c176a-f547-4206-af85-c63a4364b0d8
Locale: en_GB
Authorization: Basic aGVybWFuLmphbGlua0Bqb25hc25ldC5ubDpISjM3ODFodw==
Content-Type: application/vnd.wirelesscar.com.voc.ClientPosition.v4+json; charset=utf-8
User-Agent: Dalvik/2.1.0 (Linux; U; Android 6.0.1; MotoG3 Build/MPIS24.107-55-2-17)
Host: vocapi.wirelesscar.net
Connection: Keep-Alive
Accept-Encoding: gzip
Content-Length: 94

{"clientAccuracy":0.0,"clientLatitude":lat_coordinate,"clientLongitude":lon_coordinate}

//Received command from VoC server to VoC application

HTTP/1.1 200 OK
Server: nginx
Date: Thu, 01 Aug 2019 13:52:38 GMT
Content-Type: application/vnd.wirelesscar.com.voc.Service.v4+json;charset=utf-8
Content-Length: 317
Connection: keep-alive
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: 0
Strict-Transport-Security: max-age=31536000 ; includeSubDomains
X-XSS-Protection: 1; mode=block
X-Frame-Options: DENY
X-Content-Type-Options: nosniff
Strict-Transport-Security: max-age=31536000

{"status":"Started","statusTimestamp":"2019-08-01T13:52:38+0000","startTime":"2019-08-01T13:52:38+0000","serviceType":"RHBLF","failureReason":null,"service":"https://vocapi.wirelesscar.net/customerapi/rest/vehicles/<VIN number>/services/number_7_digits","vehicleId":"<VIN number>","customerServiceId":"number_7_digits"}

// Send command from VoC application to VoC server

GET /customerapi/rest/vehicles/<VIN number>/services/number_7_digits
HTTP/1.1
X-Client-Version: 4.4.5.21126
X-Device-Id: 55792eff9c2ac026
X-OS-Type: Android
X-App-Name: Volvo On Call
X-OS-Version: 23
X-Originator-Type: app
Accept: application/vnd.wirelesscar.com.voc.Service.v4+json; charset=utf-8
X-Request-Id: 8afe1536-8ad3-4748-873f-752c1ca478a9
Locale: en_GB
Authorization: Basic aGVybWFuLmphbGlua0Bqb25hc25ldC5ubDpISjM3ODFodw==
User-Agent: Dalvik/2.1.0 (Linux; U; Android 6.0.1; MotoG3 Build/MPIS24.107-55-2-17)
Host: vocapi.wirelesscar.net
Connection: Keep-Alive
Accept-Encoding: gzip

//Received command from VoC server to VoC application

HTTP/1.1 200 OK
Server: nginx
Date: Thu, 01 Aug 2019 13:52:44 GMT
Content-Type: application/vnd.wirelesscar.com.voc.Service.v4+json;charset=utf-8
Content-Length: 326
Connection: keep-alive
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: 0
Strict-Transport-Security: max-age=31536000 ; includeSubDomains
X-XSS-Protection: 1; mode=block
X-Frame-Options: DENY
X-Content-Type-Options: nosniff
Strict-Transport-Security: max-age=31536000

{"status":"MessageDelivered","statusTimestamp":"2019-08-01T13:52:41+0000","startTime":"2019-08-01T13:52:38+0000","serviceType":"RHBLF","failureReason":null,"service":"https://vocapi.wirelesscar.net/customerapi/rest/vehicles/<VIN number>/services/number_7_digits","vehicleId":"<VIN number>","customerServiceId":"number_7_digits"}

//Send command from VoC application to VoC server

GET /customerapi/rest/vehicles/<VIN number>/services/number_7_digits HTTP/1.1
X-Client-Version: 4.4.5.21126
X-Device-Id: 55792eff9c2ac026
X-OS-Type: Android
X-App-Name: Volvo On Call
X-OS-Version: 23
X-Originator-Type: app
Accept: application/vnd.wirelesscar.com.voc.Service.v4+json; charset=utf-8
X-Request-Id: eaf23680-a38b-4fcb-a0fc-147cb2ae1bed
Locale: en_GB
Authorization: Basic aGVybWFuLmphbGlua0Bqb25hc25ldC5ubDpISjM3ODFodw==
User-Agent: Dalvik/2.1.0 (Linux; U; Android 6.0.1; MotoG3 Build/MPIS24.107-55-2-17)
Host: vocapi.wirelesscar.net
Connection: Keep-Alive
Accept-Encoding: gzip

//Received command from VoC server to VoC application

HTTP/1.1 200 OK
Server: nginx
Date: Thu, 01 Aug 2019 13:52:54 GMT
Content-Type: application/vnd.wirelesscar.com.voc.Service.v4+json;charset=utf-8
Content-Length: 320
Connection: keep-alive
Cache-Control: no-cache, no-store, max-age=0, must-revalidate
Pragma: no-cache
Expires: 0
Strict-Transport-Security: max-age=31536000 ; includeSubDomains
X-XSS-Protection: 1; mode=block
X-Frame-Options: DENY
X-Content-Type-Options: nosniff
Strict-Transport-Security: max-age=31536000

{"status":"Successful","statusTimestamp":"2019-08-01T13:52:49+0000","startTime":"2019-08-01T13:52:38+0000","serviceType":"RHBLF","failureReason":null,"service":"https://vocapi.wirelesscar.net/customerapi/rest/vehicles/<VIN number>/services/number_7_digits","vehicleId":"<VIN number>","customerServiceId":"number_7_digits"}
-------------------------------------------------------------------------------------------------------
The lines after // are my comments.
For privacy reasons, I removed my VIN number and the lat and lon co-ordinates.
If you look to the message sequence, you see that they use a so called: "customerServiceId":"number_7_digits".
This is a seven digit number, that I have removed for privacy reasons as well.

The question is:
Is it possible to make honk and blink work?
Would be nice?

Regards,

Herman
